s32/mcux/devices/S32Z270/fsl_device_registers.h

```diff
@@ -16,7 +16,10 @@
 #if defined(CPU_S32Z270)
 
 /* CMSIS-style register definitions */
-#include "S32Z2.h"
+#include "S32Z2_PIT.h"
```

Comment:
>add essential header files to build with mcux

elaborate pls, why this is needed?

Reply:
+ In _"S32Z2.h"_ included _"S32Z2_SPI.h"_ that has the struct **SPI_Type** and the macros as same as which I has defined in  _modules\hal\nxp\s32\mcux\devices\S32Z270\S32Z270_device.h_ and **mcux dspi** is using, leads to raise error re-defined. If we want to build DSPI with mcux, we cannot include "S32Z2_SPI.h"

+ If we include <S32Z2.h>, we could add unnecessary files to build

That why I want to include the essential files

I propose 2 solutions:
+ do like which implemented
+ jus remove or guard _#include "S32Z2_SPI.h"_ in _modules\hal\nxp\s32\drivers\s32ze\BaseNXP\header\S32Z2.h_ and keep them all

Reply:
I see you have changed the approach in https://github.com/zephyrproject-rtos/hal_nxp/pull/435/commits/83f04dc0fdf09bc3e0b92bccee2810c039043cb1 
patches to the HAL must be documented in the s32/README

Reply:
I updated

Reply:
I must say that I prefer the previous approach, including only what's necessary in zephyr's soc.h, but we can do it in a separate pr that address all drivers.

Reply:
yes, we will update in the future PRs.